aws/terraform/metaflow/iam-custom-role.tf

```diff
@@ -0,0 +1,317 @@
+      "arn:${var.iam_partition}:kms:${local.aws_region}:${local.aws_account_id}:key/"
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "batch_perms" {
+  statement {
+    sid = "JobsPermissions"
+
+    effect = "Allow"
+
+    actions = [
+      "batch:TerminateJob",
+      "batch:DescribeJobs",
+      "batch:DescribeJobDefinitions",
+      "batch:DescribeJobQueues",
+      "batch:RegisterJobDefinition",
+      "batch:DescribeComputeEnvironments",
+    ]
+
+    resources = [
+      "*"
+    ]
+  }
+
+  statement {
+    sid = "DefinitionsPermissions"
+
+    effect = "Allow"
+
+    actions = [
+      "batch:SubmitJob"
+    ]
+
+    resources = [
+      module.metaflow.METAFLOW_BATCH_JOB_QUEUE,
+      "arn:${var.iam_partition}:batch:${local.aws_region}:${local.aws_account_id}:job-definition/*:*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "custom_s3_list_access" {
+  statement {
+    sid = "BucketAccess"
+
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket"
+    ]
+
+    resources = [
+      module.metaflow.metaflow_s3_bucket_arn
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "log_perms" {
+  statement {
+    sid = "GetLogs"
+
+    effect = "Allow"
+
+    actions = [
+      "logs:GetLogEvents"
+    ]
+
+    resources = [
+      "arn:${var.iam_partition}:logs:${local.aws_region}:${local.aws_account_id}:log-group:*:log-stream:*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "allow_sagemaker" {
+  statement {
+    sid = "AllowSagemakerCreate"
+
+    effect = "Allow"
+
+    actions = [
+      "sagemaker:CreateTrainingJob"
+    ]
+
+    resources = [
+      "arn:${var.iam_partition}:sagemaker:${local.aws_region}:${local.aws_account_id}:training-job/*",
+    ]
+  }
+
+  statement {
+    sid = "AllowSagemakerDescribe"
+
+    effect = "Allow"
+
+    actions = [
+      "sagemaker:DescribeTrainingJob"
+    ]
+
+    resources = [
+      "arn:${var.iam_partition}:sagemaker:${local.aws_region}:${local.aws_account_id}:training-job/*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "allow_step_functions" {
+  statement {
+    sid = "TasksAndExecutionsGlobal"
+
+    effect = "Allow"
+
+    actions = [
+      "states:ListStateMachines"
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    sid = "StateMachines"
+
+    effect = "Allow"
+
+    actions = [
+      "states:DescribeStateMachine",
+      "states:UpdateStateMachine",
+      "states:StartExecution",
+      "states:CreateStateMachine",
+      "states:ListExecutions",
+      "states:StopExecution"
+    ]
+
+    resources = [
+      "arn:${var.iam_partition}:states:${local.aws_region}:${local.aws_account_id}:stateMachine:*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "allow_event_bridge" {
+  statement {
+    sid = "RuleMaintenance"
+
+    effect = "Allow"
+
+    actions = [
+      "events:PutTargets",
+      "events:DisableRule",
+    ]
+
+    resources = [
+      "arn:${var.iam_partition}:events:${local.aws_region}:${local.aws_account_id}:rule:*",
```

Comment:
I believe this needs to be `arn:${var.iam_partition}:events:${local.aws_region}:${local.aws_account_id}:rule/*`